plugin/trino-raptor-legacy/src/main/java/io/trino/plugin/raptor/legacy/storage/RaptorStorageManager.java

```diff
@@ -469,27 +456,7 @@ private static OrcFileInfo rewriteFile(File input, File output, BitSet rowsToDel
 
     private List<ColumnInfo> getColumnInfo(OrcReader reader)
     {
-        Optional<OrcFileMetadata> metadata = getOrcFileMetadata(reader);
-        if (metadata.isPresent()) {
-            return getColumnInfoFromOrcUserMetadata(metadata.get());
-        }
-
-        // support for legacy files without metadata
-        return getColumnInfoFromOrcColumnTypes(reader.getColumnNames(), reader.getFooter().getTypes());
-    }
-
-    private List<ColumnInfo> getColumnInfoFromOrcColumnTypes(List<String> orcColumnNames, ColumnMetadata<OrcType> orcColumnTypes)
-    {
-        Type rowType = getType(orcColumnTypes, ROOT_COLUMN);
-        if (orcColumnNames.size() != rowType.getTypeParameters().size()) {
-            throw new TrinoException(RAPTOR_ERROR, "Column names and types do not match");
-        }
-
-        ImmutableList.Builder<ColumnInfo> list = ImmutableList.builder();
-        for (int i = 0; i < orcColumnNames.size(); i++) {
-            list.add(new ColumnInfo(Long.parseLong(orcColumnNames.get(i)), rowType.getTypeParameters().get(i)));
-        }
-        return list.build();
+        return getColumnInfoFromOrcUserMetadata(getOrcFileMetadata(reader));
```

Comment:
maybe inline this

Reply:
I considered that, but it's called from a different class, so this hides the internal methods.